bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c

```diff
@@ -112,21 +112,73 @@ i2c_master_bus_handle_t bsp_i2c_get_handle(void)
     return i2c_handle;
 }
 
-esp_err_t bsp_sdcard_mount(void)
+sdmmc_card_t *bsp_sdcard_get_handle(void)
+{
+    return bsp_sdcard;
+}
+
+sdmmc_host_t *bsp_sdcard_get_host(void)
+{
+    static sdmmc_host_t host_config = SDMMC_HOST_DEFAULT();
+    host_config.slot = SDMMC_HOST_SLOT_0;
+    host_config.max_freq_khz = SDMMC_FREQ_HIGHSPEED;
+
+    return &host_config;
```

Comment:
Not sure this is a good idea to return the address of the static variable. The caller may want to change the flags, and this will modify the static variable.

Can you instead let the caller declare the config and pass the pointer into this function? This function can then fill the structure. After that the caller can still modify some fields. There will be no static RAM usage this way.

Reply:
Thank you, you are right. Other functions can stay as they are, or should they be same?

Reply:
I think bsp_sdcard_sdspi_get_slot should be changed the same way, because you are modifying the structure depending on the spi_host argument.

Reply:
Then I'll change everything so that the API is the same :-)